pypesto/visualize/misc.py

```diff
@@ -225,3 +236,51 @@ def process_start_indices(start_indices: Union[int, Iterable[int]],
                      start_index < max_length]
 
     return start_indices
+
+
+def rgba2rgb(fg: RGB_RGBA, bg: RGB_RGBA = None) -> RGB:
+    """Combine two colors, removing transparency.
+
+    Parameters
+    ----------
+    fg:
+        Foreground color.
+    bg:
+        Background color.
+
+    Returns
+    -------
+    The combined color.
+    """
+    if bg is None:
+        bg = RGBA_WHITE
+    if len(bg) == LEN_RGBA:
+        # return foreground if background is fully transparent
+        if bg[RGBA_ALPHA] == RGBA_MIN:
+            return fg
+    else:
+        if len(bg) != LEN_RGB:
+            raise IndexError(
+                'A background color of unexpected length was provided: {bg}'
+            )
+        bg = (*bg, RGBA_MAX)
+
+    # return the foreground color if has no transparency
+    if len(fg) == LEN_RGB or fg[RGBA_ALPHA] == RGBA_MAX:
+        return fg
+    if len(fg) != LEN_RGBA:
+        raise IndexError(
+            'A foreground color of unexpected length was provided: {fg}'
+        )
+
+    def apparent_color(fg_i: float, bg_i: float) -> float:
+        """Porter and Duff equations."""
```

Comment:
just out of interest (and maybe to annotate if possible: wtf are those equations?

Reply:
Added an explanation in the docstring (just used for alpha compositing)